block/src/Pos/Block/Network/Logic.hs

```diff
@@ -97,7 +95,9 @@ instance Exception BlockNetLogicException where
 -- progress and 'ncRecoveryHeader' is full, we'll be requesting blocks anyway
 -- and until we're finished we shouldn't be asking for new blocks.
 triggerRecovery
-    :: BlockWorkMode ctx m
+    :: ( BlockWorkMode ctx m
+       , HasProtocolMagic
+       )
```

Comment:
If we're doing a refactor here anyway, it's just as easy to take `ProtocolMagic` as a parameter rather than the magical `HasProtocolMagic` constraint.